sdk/communication/azure-communication-email/azure/communication/email/_email_client.py

```diff
@@ -0,0 +1,103 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from uuid import uuid4
+from azure.core.tracing.decorator import distributed_trace
+from ._shared.utils import parse_connection_str, get_current_utc_time
+from ._shared.policy import HMACCredentialsPolicy
+from ._generated._azure_communication_email_service import AzureCommunicationEmailService
+from ._version import SDK_MONIKER
+from ._generated.models import SendEmailResult, SendStatusResult, EmailMessage
+
+class EmailClient(object): # pylint: disable=client-accepts-api-version-keyword
+    """A client to interact with the AzureCommunicationService Email gateway.
+
+    This client provides operations to send an email and monitor its status.
+
+    :param str endpoint:
+        The endpoint url for Azure Communication Service resource.
+    :param TokenCredential credential:
+        The TokenCredential we use to authenticate against the service.
+    """
+    def __init__(
+            self,
+            endpoint, # type: str
+            credential, # type: TokenCredential
+            **kwargs # type: Any
+        ):
+        # type: (...) -> None
+        authentication_policy = HMACCredentialsPolicy(endpoint, credential)
+
+        self._generated_client = AzureCommunicationEmailService(
+            endpoint,
+            authentication_policy=authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            **kwargs
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str, # type: str
+        **kwargs # type: Any
+    ): # type: (...) -> EmailClient
+        """Create EmailClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Communication Service resource.
+        :returns: Instance of EmailClient.
+        :rtype: ~azure.communication.EmailClient
+        """
+        endpoint, access_key = parse_connection_str(conn_str)
+
+        return cls(endpoint, access_key, **kwargs)
+
+    @distributed_trace
+    def send(
+        self,
+        email_message, # type: EmailMessage
+        **kwargs # type: Any
+    ): # type: (...) -> SendEmailResult
+        """Queues an email message to be sent to one or more recipients.
+
+        :param email_message: The message payload for sending an email.
+        :type email_message: ~azure.communication.email.models.EmailMessage
+        :return: SendEmailResult
+        :rtype: ~azure.communication.email.models.SendEmailResult
+        """
+
+        return self._generated_client.email.send(
+            repeatability_request_id=uuid4(),
```

Comment:
How is this `repeatability_request_id` used? Is it a problem that this is never exposed to customers?

Reply:
For now, we're just generating a random GUID and using that in the request. Our team is currently working on implementing retry logic with this repeatability_request_id. This will be implemented later.